### DISCUSSION TITLE
Diferença de saída entre console.log(obj) e console.log(`${obj}`)

### DISCUSSION THREAD

**Discussion by Cyber-Phantom** [frontendbr/forum#2432] [Perguntas e Respostas]
Pessoal, tô com uma dúvida que não consigo resolver (sou iniciante em JS):

## COMANDOS

```
console.log(ctx)
console.log(`Passos 1, 2, 3: ${ctx}`)
```
## SAÍDAS
```
{ valor1: 'mid1', valor2: 'mid2', valor3: 'mid3' }
Passos 1, 2, 3: [object Object]
```
Qual comando/função usar para que a segunda saída ( `[object Object]` ) seja no formato da primeira ( `{ valor1: 'mid1', valor2: 'mid2', valor3: 'mid3' }` )? sem transformar em JSON (`JSON.stringify(ctx)`). Ou seja, gostaria de obter o conteúdo como na primeira saída.<br>

Agradeço desde já qq ajuda.

**Comment by igorlvicente** [comment]
Não sei te responder, mas vou te sugerir o que procurar. Procure como o `console.log` trata os seus argumentos. Normalmente os esses comandos de print fazem algo com o objeto, tipo chamar `.to_s` (no Ruby).
Depois de você encontrar o que o `console.log` está fazendo com o `obj`, você procura se dá pra alterar algo no objeto que faça com que ele printe bonitinho (no exemplo do ruby, seria alterar o método `.to_s` na classe).

**Reply by igorlvicente** [reply]
Onde procurar:
https://developer.mozilla.org/en-US/docs/Web/API/console/log
https://developer.mozilla.org/en-US/docs/Web/API/console#outputting_text_to_the_console

**Reply by igorlvicente** [reply]
Aliás, erro meu. Você não deve procurar o que o `console.log` faz porque ele já está recebendo o objeto como string. O que você deve procurar, na verdade, é como o Javascript transforma um objeto em string, que é o que acontece quando você usa `` `${obj}` ``

**Reply by igorlvicente** [reply]
Onde procurar:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

**Comment by Cyber-Phantom** [comment]
Valeuuu, @igorlvicente !

Grato aí pelos caminhos. Vou fuçar esse negócio. Isso me deixou intrigado. 
Deve haver alguma função em JS para imprimir o obj na template sem desconfigurar a saída...

Ehhh, vida de garoto de programa... sempre correndo atrás... kkkk

Gratidão, cara!

**Comment by Cyber-Phantom** [comment]
@igorlvicente !!!!

Achei  a parada, meu camarada!! Tô repassando aqui pra ti.

## COMANDOS

```
console.log(ctx)
console.info('Passos 1, 2, 3: ', ctx)
console.log(`Passos 1, 2, 3: ${JSON.stringify(ctx)}`)
```

## SAÍDAS

```
{ valor1: 'mid1', valor2: 'mid2', valor3: 'mid3' }
Passos 1, 2, 3:  { valor1: 'mid1', valor2: 'mid2', valor3: 'mid3' }
Passos 1, 2, 3: {"valor1":"mid1","valor2":"mid2","valor3":"mid3"}
```

@igorlvicente , veja o SEGUNDO comando!

E encontrei a partir de uma das tuas referências, véi. Show de bola.

https://developer.mozilla.org/en-US/docs/Web/API/console#outputting_text_to_the_console

Dei de cara com isso:

![1](https://github.com/frontendbr/forum/assets/132688085/e4e7c2ad-5625-48d8-a122-cbfa63f86917)

Valeuuuu mesmo, meu camarada! Tmj.

**Reply by igorlvicente** [reply]
Se era isso que você queria, viva 🥳 
E bem descrito o seu experimento (esse último post). Obrigado por isso.
